app/components/Account/AccountVoting.jsx

```diff
@@ -593,29 +580,161 @@ class AccountVoting extends React.Component {
+        const hasProxy = !!proxy_account_id; // this.props.account.getIn(["options", "voting_account"]) !== "1.2.5";
+        const {globalObject, account} = this.props;
+        const {totalBudget, workerBudget} = this._getBudgets(globalObject);
+
+        const actionButtons = this._getActionButtons();
+
+        const proxyInput = this._getProxyInput(accountHasProxy);
+
+        const hideLegacy = this._getHideLegacyOptions();
+
+        const onFilterChange = this.handleFilterChange.bind(this);
+        const validateAccountHandler = this.validateAccount.bind(
+            this,
+            WITNESSES_KEY
+        );
+        const addWitnessHandler = this.onAddItem.bind(this, WITNESSES_KEY);
+        const removeWitnessHandler = this.onRemoveItem.bind(
+            this,
+            WITNESSES_KEY
+        );
+        const setWorkersLength = this.setWorkersLength.bind(this);
+        const onChangeVotes = this.onChangeVotes.bind(this);
+        const getWorkerArray = this._getWorkerArray.bind(this);
+        const addCommitteeHandler = this.onAddItem.bind(this, COMMITTEE_KEY);
+        const removeCommitteeHandler = this.onRemoveItem.bind(
+            this,
+            COMMITTEE_KEY
+        );
+
+        const onTabChange = value => {
+            this.props.history.push(value);
+        };
+
+        return (
+            <div className="grid-content no-padding page-layout ">
+                <div className="main-content content-block small-12 voting">
+                    <div className="padding">
+                        <div>
+                            <Translate content="voting.title" component="h1" />
+                            <Translate
+                                content="voting.description"
+                                component="p"
+                            />
+                        </div>
+                        <div className="proxy-row">
+                            {proxyInput}
+                            {actionButtons}
+                        </div>
+                    </div>
+
+                    <Tabs
+                        activeKey={this.props.location.pathname}
+                        animated={false}
+                        style={{
+                            display: "table",
+                            height: "100%",
+                            width: "100%"
+                        }}
+                        onChange={onTabChange}
+                    >
+                        {this.state.tabs.map(tab => {
+                            const TabContent = tab.content;
+
+                            return (
+                                <Tabs.TabPane
+                                    key={tab.link}
+                                    tab={counterpart.translate(tab.translate)}
+                                >
+                                    <div className="padding">
+                                        <TabContent
```

Comment:
Please check the scope of all those props.

Where are they properly encapsulated? What needs to be in AccountVoting, what can also be only in the sub-components?

Reply:
Found that it can be simplified by extraction of several workers related properties. Some of props are still related only to certain tabs, but those are quite hard to extract and will impact a parent component havily since it has now all the state management merged in it. 
This can be done, but will require ~4h to refactor witnesses, workers and committee components.

Reply:
Yea, the voting logic should be in the parent just like before. I would have hoped that the parent only needs to store the voting_ids and onaction handlers to allow manipulation of voting_ids. 

Leave as is, I will try to evaluate more if changes are necessary.